### PR TITLE
Move api codec into api module. Add a client implementation

### PR DESCRIPTION
**Pull request by flxo**

### CODE REVIEW

File: northstar/src/api/codec.rs [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+    type Error = io::Error;
+
+    fn decode(&mut self, src: &mut bytes::BytesMut) -> Result<Option<Self::Item>, Self::Error> {
+        if let Some(position) = src.iter().position(|b| *b == b'\n') {
+            let buf = src.split_to(position);
+            // Consume the newline
+            src.advance(1);
+            match serde_json::from_slice::<model::Message>(&buf) {
+                Ok(message) => Ok(Some(message)),
+                Err(e) => Err(io::Error::new(ErrorKind::Other, e)),
```
Comment (marcmo) [inline northstar/src/api/codec.rs:53]: better use `ErrorKind::InvalidData`

File: northstar/src/api/codec.rs [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+    }
+
+    fn poll_flush(mut self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Result<(), Self::Error>> {
+        Pin::new(&mut self.framed).poll_flush(cx)
+    }
+
+    fn poll_close(mut self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Result<(), Self::Error>> {
+        Pin::new(&mut self.framed).poll_close(cx)
+    }
+}
```
Comment (marcmo) [inline northstar/src/api/codec.rs:144]: The `AsyncWrite`, `AsyncRead`, `Sink` and `Stream` implementation looks a lot like boilerplate code. My understanding of the codec package was that you can use `FramedRead` and `FramedWrite` to achieve this.

Comment (flxo) [reply northstar/src/api/codec.rs:144]: `FramedWrite` and `AsyncWrite` work different. If we want to use `FramedWrite` (mean `Sink`) for sending the install stream we need to introduce another enum variant for `Framed::Item` and chunk manually. This is safe but probably slower than this direct access to the wrapped `AsyncWrite`. The intention to newtype `Framed` is to use it as `Stream`, `Sink` *and* `AsyncWrite`.

File: northstar/src/api/codec.rs (hunk 1 above)
Comment (flxo) [reply northstar/src/api/codec.rs:53]: With `ErrorKind::Invalid` we'd discard the error information from `serde`.

Comment (marcmo) [reply northstar/src/api/codec.rs:53]: why is this?
`Err(e) => Err(io::Error::new(ErrorKind::InvalidData, e)),`
what information is lost here?

Comment (flxo) [reply northstar/src/api/codec.rs:53]: Ah. Its a `ErrorKind`. My bad. This makes sense. I'll change that.

**Comment by marcmo** [comment]
would be nice to add some info about this change to our changelog

**Comment by flxo** [comment]
> would be nice to add some info about this change to our changelog

Done.

**Pull request merged**